lib/portlayer/event/event_manager.go

```diff
@@ -15,38 +15,31 @@
 package event
 
 import (
-	"github.com/vmware/vic/pkg/vsphere/session"
+	"github.com/vmware/vic/lib/portlayer/event/collector"
+	"github.com/vmware/vic/lib/portlayer/event/events"
 )
 
+// EventManager will provide a basic event pub/sub implementation
 type EventManager interface {
 
-	// AddMonitoredObject will add the object for event listening
-	AddMonitoredObject(ref string) error
+	// RegisterCollector a collector with the manager
+	RegisterCollector(collector.Collector)
 
-	// RemoveMonitoredObject will remove the object from event listening
-	RemoveMonitoredObject(ref string)
+	// Collectors returns registered collectors
+	Collectors() map[string]collector.Collector
 
-	// Register will for event callbacks
-	Register(caller string, callback func(Event, *session.Session))
+	// Subscribe for event callbacks
+	Subscribe(eventTopic string, caller string, callback func(events.Event))
```

Comment:
Yeah...agree would be more flexible...I'd open an issue as a suggested enhancement